src/objects/zcl_abapgit_object_iaxu.clas.abap

```diff
@@ -0,0 +1,305 @@
+    w3_api_save( io_xml_api = lo_xml_api ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~changed_by.
+    rv_user = c_user_unknown. " todo
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~delete.
+
+    DATA: lo_xml_api TYPE REF TO cl_w3_api_xml3,
+          ls_name    TYPE iacikeyt,
+          lx_root    TYPE REF TO zcx_abapgit_exception.
+
+
+    ls_name = ms_item-obj_name.
+
+    w3_api_load( EXPORTING is_name    = ls_name
+                 IMPORTING eo_xml_api = lo_xml_api ).
+
+    w3_api_set_changeable( io_xml_api    = lo_xml_api
+                           iv_changeable = abap_true ).
+
+    w3_api_delete( io_xml_api = lo_xml_api ).
+
+    w3_api_save( io_xml_api = lo_xml_api ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~deserialize.
+
+    DATA: ls_attr TYPE w3tempattr.
+
+
+    io_xml->read( EXPORTING iv_name = 'ATTR'
+                  CHANGING  cg_data = ls_attr ).
+
+    ls_attr-devclass = iv_package.
+
+    IF zif_abapgit_object~exists( ) = abap_true.
+      zif_abapgit_object~delete( ).
+    ENDIF.
+
+    save( is_attr = ls_attr ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~exists.
+
+    DATA: ls_name TYPE iacikeyt.
+
+
+    ls_name = ms_item-obj_name.
+
+    TRY.
+        w3_api_load( is_name = ls_name ).
+
+        rv_bool = abap_true.
+
+      CATCH zcx_abapgit_object_not_found.
+        rv_bool = abap_false.
+
+    ENDTRY.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~get_comparator.
+    RETURN.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~get_deserialize_steps.
+    APPEND zif_abapgit_object=>gc_step_id-abap TO rt_steps.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~get_metadata.
+    rs_metadata = get_metadata( ).
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~is_active.
+    rv_active = is_active( ).
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~is_locked.
+    rv_is_locked = abap_false.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~jump.
+
+    CALL FUNCTION 'RS_TOOL_ACCESS'
+      EXPORTING
+        operation   = 'SHOW'
+        object_name = ms_item-obj_name
+        object_type = ms_item-obj_type.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~serialize.
+
+    DATA: ls_attr TYPE w3tempattr.
+
+
+    IF zif_abapgit_object~exists( ) = abap_false.
+      RETURN.
+    ENDIF.
+
+    read( IMPORTING es_attr = ls_attr ).
+
+    io_xml->add( iv_name = 'ATTR'
+                 ig_data = ls_attr ).
+
+  ENDMETHOD.
+
+
+  METHOD w3_api_load.
+
+    cl_w3_api_xml3=>load(
+      EXPORTING
+        p_xml_name          = is_name
+      IMPORTING
+        p_attributes        = es_attr
+      EXCEPTIONS
+        object_not_existing = 1
+        permission_failure  = 2
+        data_corrupt        = 3
+        error_occured       = 4
+        OTHERS              = 5 ).
+
+    IF sy-subrc = 1.
+      zcx_abapgit_object_not_found=>raise( |Error from w3_api_xml3~load subrc={ sy-subrc }| ).
+    ELSEIF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise( |Error from w3_api_xml3~load subrc={ sy-subrc }| ).
+    ENDIF.
```

Comment:
Hm, we already have `zcx_abapgit_not_found` exception class. Is the new one `ZCX_ABAPGIT_OBJECT_NOT_FOUND` really needed? I would prefer to use plain `zcx_abapgit_exception` as it simplifies the logic. See bug above.

Reply:
If I would use ZCX_ABAPGIT_EXCEPTION I couldn't catch it in the method for the existence check and set the boolean to abap_false. Additionally ZCX_ABAPGIT_NOT_FOUND doesn't inherit from ZCX_ABAPGIT_EXCEPTION and therefore hasn't a method RAISE which would allow me to throw a message up to the GUI.

Reply:
Ah, now I understand... Method Raise in ZCX_ABAPGIT_OBJECT_NOT_FOUND doesn't throw ZCX_ABAPGIT_OBJECT_NOT_FOUND but ZCX_ABAPGIT_EXCEPTION and therefore isn't being catched. Ok, I'll change it according to your suggestions.